### ISSUE TITLE
swc/jest & inversify: Test execution error for classes with @injectable decorator

### ISSUE THREAD

**Issue by hcoggins** [swc-project/swc#3520]
### Describe the bug

Executing a test for a class with an @injectable decorator that also calls its own static method to initialize readonly attributes fails with an error.

### Input code

```typescript
import { injectable } from 'inversify'

@injectable()
export class Example {
    private readonly attribute = Example.staticMethod()

    public static staticMethod(): boolean {
        return true
    }
}
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "decorators": true,
      "tsx": true
    },
    "keepClassNames": true,
    "transform": {
      "legacyDecorator": true,
      "decoratorMetadata": true
    }
  },
  "module": {
    "type": "commonjs"
  }
}
```

### Playground link

https://play.swc.rs/?version=1.2.135&code=H4sIAAAAAAAAA1WNSw7CMAxE9zmFd4UNB0BCYsOSQzipK4zyk%2BNUrarendCCELOcmTfDISdRWIDjk5yi9QQrDJICdBxHksLD3Blz%2FeWHo6Fpo5zHUuA2YcgNWww0ZeERlUAI%2BxT9DKgqbGuzLt%2FqqSgquzvpI%2FVtbger9exgj%2BC%2FcQabkieMn5O3hLRKBJVKm7ea9QUYkrkQzQAAAA%3D%3D&config=H4sIAAAAAAAAA0XOQQ7DIAwEwL%2F4zAu4Jtf2Dxa4VVPAyHakoIi%2FNyhNe9vD7mh3WDSA36GiKMlI2orhBh6sVdIgr2rgIFJgQWNR8CYrOTDdztgdvInqlFD1jpn%2BDcGiD5Y82ERPDG2%2BmKvzc29kGNHwax5o5rgmGtvx5DgUOGcui0LvH6Z1WjO4AAAA

### Expected behavior

I would expect the test to execute.

### Actual behavior

The following error is thrown:

`ReferenceError: _class is not defined`

This occurs upon initialization of the attribute.

### Version

swc/jest: 0.2.17; inversify: 6.0.1

### Additional context

Thanks in advance for taking the time to look into this!

**Comment by petrzjunior** [comment]
Duplicate of https://github.com/swc-project/swc/issues/2224, the problem somehow related to property initializers in decorated classes. Having no knowledge of the architecture of this project project whatsoever, I tired to replicate it in a unit test inside the swc transpiler, but I got back a correct code. So the error must be somewhere in the emit phase.

**Comment by petrzjunior** [comment]
@hcoggins #2224 was fixed in 1.2.145 and seems to solve this issue as well.

**Comment by kdy1** [comment]
Thank you!
